app/src/components/experiment/ExperimentCompareDetails.tsx

```diff
@@ -176,7 +176,17 @@ export function ExperimentCompareDetails({
   const referenceOutput = data.example.revision?.referenceOutput;
   const experimentRuns = data.example.experimentRuns?.edges;
   const experiments = data.dataset.experiments?.edges;
-  const annotationSummaries = data.dataset.experimentAnnotationSummaries;
+  const annotationSummaries = useMemo(() => {
+    return (
+      data.dataset.experimentAnnotationSummaries?.filter((summary) =>
+        experimentRuns?.some((run) =>
+          run.run.annotations?.edges.some(
+            (edge) => edge.annotation.name === summary.annotationName
+          )
+        )
+      ) ?? []
+    );
+  }, [data.dataset.experimentAnnotationSummaries, experimentRuns]);
```

Comment:
Maybe drop a one-line comment above this to explain this filter since it is a bit complex